### ISSUE TITLE
How do I register data types with ParquetRowWriter?

### ISSUE THREAD

**Issue by soerenwolfers** [G-Research/ParquetSharp#375]
I'm using 
```
ParquetRowWriter<AnonymousType>.WriteRow(new {X = x; Y = y));
```
to write parquet files. This is awesome, but I quite commonly accidentally provide values that parquet doesn't know about.

Is it possible to inform ParquetRowWriter how to serialize a given datatype? 

Ideally, I'd like to be able to 
```
public double MyXConverter(X x){
    return x.AsDouble();
}
public string MyYConverter(Y y){
    return y.Value;
}
var myParquetRowWriter = new ParquetRowWriter<AnonymousType>(..., new[] {MyXConverter, MyYConverter});
...
myParquetRowWriter.WriteRow(new {X = x; Y = y));
```
which would allow me to write a tiny wrapper that always provides my most commonly required converters and hopefully never be caught out again.

If this is hard, I'd be equally happy to register the converters in a static fashion rather than on a ParquetRowWriter instance.

**Comment by adamreeve** [comment]
Hi @soerenwolfers, thanks for raising this issue. We do support custom conversion logic through the use of custom type factories, which is documented at https://github.com/G-Research/ParquetSharp/blob/master/docs/TypeFactories.md, but this is currently only supported when using the lower level API and not the row oriented API.

It looks like it should be fairly trivial to add this feature though by extending the `ParquetRowWriter` and `ParquetRowReader` classes to allow setting the `LogicalTypeFactory`, `LogicalWriteConverterFactory` and `LogicalReadConverterFactory`.

**Comment by jescalada** [comment]
Hi, I'd like to take a shot at this issue!

I am currently working on another documentation PR, but will hopefully start this one next week.

**Comment by adamreeve** [comment]
This has been fixed in #393 and will be included in version 14
